ui/pages/setup_passkey.tsx

```diff
@@ -80,27 +89,107 @@ const SetupPasskey: NextPage = () => {
           },
         })
         .then(() => {
-          window.location.href = "./setup_passkey";
+          setFlow(undefined);
+          setLoadingKeysFlow(flow);
         })
         .catch(handleFlowError("settings", setFlow));
     }
     return Promise.resolve();
   };
 
+  const renderFlow = flow ?? loadingKeysFlow;
+  const existingKeys: UiNode[] = [];
+
   const webauthnFlow = {
-    ...flow,
+    ...renderFlow,
     ui: {
-      ...flow?.ui,
-      nodes: flow?.ui.nodes.filter(({ group }) => {
-        return group === "webauthn" || group === "default";
-      }),
+      ...renderFlow?.ui,
+      nodes: renderFlow?.ui.nodes
+        .filter((node) => {
+          if (node.meta.label?.id === ORY_LABEL_ID_REMOVE_SECURITY_ID) {
+            existingKeys.push(node);
+            return false;
+          }
+          return node.group === "webauthn" || node.group === "default";
+        })
+        .map((node) => {
+          if (node.meta.label?.id === ORY_LABEL_ID_NAME_OF_THE_SECURITY_KEY) {
+            node.meta.label.text = "Security key name";
+          }
+          return node;
+        }),
     },
   } as SettingsFlow;
 
   return (
-    <PageLayout title="Set up a passkey login method">
-      {flow ? (
-        <Flow onSubmit={handleSubmit} flow={webauthnFlow} />
+    <PageLayout title="Add a security key">
+      {webauthnFlow ? (
+        <>
+          <Flow onSubmit={handleSubmit} flow={webauthnFlow} />
+          {existingKeys.length > 0 && (
+            <>
+              <h2 className="p-heading--4">Security keys</h2>
+
+              {loadingKeysFlow && <Spinner />}
+              {!loadingKeysFlow && (
+                <>
+                  {existingKeys.map((item, k) => {
+                    let displayName = "";
+                    if (
+                      item.meta.label?.context &&
+                      "display_name" in item.meta.label.context
+                    ) {
+                      displayName = item.meta.label.context
+                        .display_name as string;
+                    }
+                    let addedAt = "";
+                    if (
+                      item.meta.label?.context &&
+                      "added_at_unix" in item.meta.label.context
+                    ) {
+                      const date = new Date(
+                        (item.meta.label.context.added_at_unix as number) *
+                          1000,
+                      );
+                      addedAt = date.toLocaleDateString("en-US", {
+                        year: "numeric",
+                        month: "short",
+                        day: "2-digit",
+                      });
+                    }
+
+                    return (
+                      <div className="row security-key-row" key={k}>
```

Comment:
is it a pattern for this project to use vanilla classes directly?

Reply:
whoops. started using the react-components one.